src/tools/perf/lib/libperf.c

```diff
@@ -1174,7 +1184,7 @@ static void ucp_perf_test_destroy_eps(ucx_perf_context_t* perf)
 
             if (UCS_PTR_IS_PTR(req)) {
                 do {
-                    ucp_worker_progress(perf->ucp.tctx[i].perf.ucp.worker);
+                    ucp_perf_worker_progress(perf);
```

Comment:
what does it fix?

Reply:
need to progress all threads when doing a barrier or closing all EPs.
before the fix, only the first thread was progressed

Reply:
yes, one process moved to destroy the 2nd EP, while another process is flushing the 1st EP.
So, we should progress the 1st worker in the first process to send ACK message to the 1st EP of the second process.

Reply:
so can we progress all "thread_count" close reqs in parallel? like we do in OMPI for example

Reply:
then each thread should take care of it closing. in current perftest implementation, master thread closes (one by one) all EPs and progress workers.

Reply:
i mean still progress in master thread, but first start all close operations and then do the progress calls